zenml/cli/pipeline.py

```diff
@@ -31,3 +37,106 @@ def set_metadata_store():
     click.echo('Comparing pipelines in repo: Starting app..')
     repo: Repository = Repository.get_instance()
     repo.compare_pipelines()
+
+
+@pipeline.command('list')
+def list_pipelines():
+    """Lists pipelines in the current repository."""
+    try:
+        repo: Repository = Repository.get_instance()
+    except Exception as e:
+        error(e)
+        return
+
+    pipelines = repo.get_pipelines()
+
+    names = [p.name for p in pipelines]
+    types = [p.PIPELINE_TYPE for p in pipelines]
+    statuses = [p.get_status() for p in pipelines]
+    cache_enabled = [p.enable_cache for p in pipelines]
+    filenames = [p.file_name for p in pipelines]
+
+    headers = ["name", "type", "cache enabled", "status", "file name"]
+
+    click.echo(tabulate(zip(names, types, cache_enabled, statuses, filenames),
+                        headers=headers))
+
+
+@pipeline.command('get')
+@click.argument('pipeline_id')
+def get_pipeline_by_id(pipeline_id: Text):
+    """
+    Gets pipeline from current repository by matching a (partial) identifier
+    against the pipeline yaml name.
+
+    """
+    config = return_pipeline_config_by_id(pipeline_id)
+
+    if config is not None:
+        click.echo(to_pretty_string(config))
+
+
+@pipeline.command('run')
+@click.argument('pipeline_id')
+@click.option("--metadata_store", default=None,
+              help="Path to a custom metadata store to use.")
+@click.option("--artifact_store", default=None,
+              help="Path to a custom metadata store to use.")
+def run_pipeline_by_id(pipeline_id: Text,
```

Comment:
Hmm unfortuantely again this is a concept that I dont think we should have. In ZenML, unlike Core Engine, we do not have a `run_by_id` concept. Perhaps a `run_by_name`  concept still makes sense, but even there I would argue the command I would support is:

`zenml pipeline run /path/to/config.yaml`

Perhaps we can discuss this via discord once you get to it

Reply:
Reworked logic to pass in a yaml filename and instantiate and run the pipeline from the config in commit f1bc3bb.